salt/utils/pyobjects.py

```diff
@@ -332,6 +330,23 @@ def __set_attributes__(cls):
 
                 match_groups[grain][match][name] = filt.__dict__[name]
 
+        if hasattr(cls, 'priority'):
```

Comment:
Since you changed the above line to be "EAFP", why not here too?

```
try:
    new_match_groups = OrderedDict([])
    for item in cls.priority:
        try:
            new_match_groups[item] = match_groups.pop(item)
        except AttributeError:
            log.error('pyobjects: priority must be an iterable')
        except KeyError:
            pass
    match_groups = new_match_groups
except TypeError:
    log.error('pyobjects: priority must be an iterable')
except AttrbuteError:
    pass
```

Bike shedding at this point, but figured I'd throw it out :)